File: app/models/list.rb
# == Schema Information
#
# Table name: lists
#
#  id         :integer          not null, primary key
#  board_id   :integer          not null
#  title      :string(255)
#  open       :boolean          default(TRUE), not null
#  created_at :datetime         not null
#  updated_at :datetime         not null
#  position   :integer
#

class List < ActiveRecord::Base
  default_scope { order("lists.position") }
  acts_as_list

  validates_uniqueness_of :slug, scoped_to: :board_id
  belongs_to :board
  has_many :cards, dependent: :destroy

  def as_json(options = {})
    super(options.merge(include: :cards))
  end

end

File: frontend/app/models/list.js
import DS from 'ember-data';

export default DS.Model.extend({
  boardId: DS.attr(),
  title: DS.attr('string'),
  slug: DS.attr('string'),
  description: DS.attr('string'),
  open: DS.attr('boolean'),
  position: DS.attr(),
  cardTemplate: DS.attr('string'),
  createdAt: DS.attr('date'),
  updatedAt: DS.attr('date'),
  board: DS.belongsTo('board'),
  cards: DS.hasMany('cards')
});
